ocean/src/clans.rs

```diff
@@ -1,39 +1,58 @@
 
+use std::collections::HashMap;
+
 #[derive(Debug)]
 pub struct ClanSystem {
-    // TODO: add necessary fields
+    pub clans: HashMap<String, Vec<String>>,
 }
 
 impl ClanSystem {
     pub fn new() -> ClanSystem {
-        unimplemented!();
+        ClanSystem {
+            clans: HashMap::new(),
+        }
     }
 
     /**
      * Returns a list of the names of the clan members for the given clan id.
      */
     pub fn get_clan_member_names(&self, clan_id: &str) -> Vec<String> {
-        unimplemented!();
+        self.clans.get(clan_id).cloned().unwrap_or_else(Vec::new)
     }
 
     /**
      * Returns the number of clans currently in existence.
      */
     pub fn get_clan_count(&self) -> usize {
-        unimplemented!();
+        self.clans.len()
+    }
+    
+
+    pub fn add_member(&mut self, clan_id: &str, crab_name: String) {
+        self.clans.entry(clan_id.to_string())
```

Comment:
We may want to consider leaving a comment here, explaining how these functions chain together to achieve the desired behavior.